src/Tizen.NET.MaterialComponents/Components/MAppbar.cs

```diff
@@ -0,0 +1,261 @@
+        Box _mainContainer;
+        EvasObject _main;
+
+        string _title;
+        string _navigationItem;
+        string _primaryItem;
+        string _secondaryItem;
+        string _overflowItem;
+
+        MButton _navigationButton;
+        MButton _primaryButton;
+        MButton _secondaryButton;
+        MButton _overflowButton;
+
+        Image _navigationImage;
+        Image _primaryImage;
+        Image _secondaryImage;
+        Image _overflowImage;
+
+        public MAppbar(EvasObject parent) : base(parent)
+        {
+            _parent = parent;
+            _naviFrame = new Naviframe(parent)
+            {
+                AlignmentY = -1,
+                AlignmentX = -1,
+                WeightY = 1,
+                WeightX = 1
+            };
+            _naviFrame.Show();
+            _mainContainer = new Box(parent);
+            _naviItem = _naviFrame.Push(_mainContainer, _title, Styles.Material);
+
+            PackEnd(_naviFrame);
+
+            MColors.AddColorSchemeComponent(this);
+        }
+
+        public EvasObject Main
+        {
+            get
+            {
+                return _main;
+            }
+            set
+            {
+                _main = value;
+                UpdateMain();
+            }
+        }
+
+        public string Title
+        {
+            get
+            {
+                return _title;
+            }
+            set
+            {
+                _title = value;
+                UpdateTitle();
+            }
+        }
+
+        public string NavigationItem
+        {
+            get
+            {
+                return _navigationItem;
+            }
+            set
+            {
+                _navigationItem = value;
+                UpdateItem(Parts.NaviItem.NavigationItem, _navigationItem, ref _navigationButton, ref _navigationImage);
+            }
+        }
+
+        public string PrimaryItem
+        {
+            get
+            {
+                return _primaryItem;
+            }
+            set
+            {
+                _primaryItem = value;
+                UpdateItem(Parts.NaviItem.PrimaryItem, _primaryItem, ref _primaryButton, ref _primaryImage);
+            }
+        }
+
+        public string SecondaryItem
+        {
+            get
+            {
+                return _secondaryItem;
+            }
+            set
+            {
+                _secondaryItem = value;
+                UpdateItem(Parts.NaviItem.SecondaryItem, _secondaryItem, ref _secondaryButton, ref _secondaryImage);
+            }
+        }
+
+        public string OverflowItem
+        {
+            get
+            {
+                return _overflowItem;
+            }
+            set
+            {
+                _overflowItem = value;
+                UpdateItem(Parts.NaviItem.OverflowItem, _overflowItem, ref _overflowButton, ref _overflowImage);
+            }
+        }
+
+        public event EventHandler NavigationItemClicked
```

Comment:
It looks like the order has been reversed. 

What about add invoke code insider the button event handler as below?
```cs
public void OnNavigationButtonClicked()
{
    NavigationItemClicked?.Invoke(this, EventArgs.Empty);
}
```

Reply:
Done